### ISSUE TITLE
Fix up source code comments

### ISSUE THREAD

**Issue by bkircher** [gridscale/gsclient-go#184]
Cool thing is that gsclient-go documentation coverage is quiet good. But there are some corners that could need some polishing.

Go through package and docs for public types and functions in gsclient-go and follow suggestions in [this](https://blog.golang.org/godoc) article. Also make sure every package has a dedicated doc string.

![image](https://user-images.githubusercontent.com/536464/108324795-a9236880-71c8-11eb-81e1-782be1f922ea.png)

**Comment by nvthongswansea** [comment]
@bkircher Could you give an example of comment that needs to be polished, please?

**Comment by bkircher** [comment]
If you run godoc locally and compare the docs with what is in the, say, standard libs you see quiet a few differences. I think all is outlined in the article linked above.

**Comment by nvthongswansea** [comment]
I've spent hours to get this godoc running. Luckily, it runs perfectly at last. The doc of gsclient-go looks ok to me 
![image](https://user-images.githubusercontent.com/9968099/110796548-5e04ee80-8278-11eb-9363-ca13ae605425.png)
**Remember**: the package name is `github.com/gridscale/gsclient-go/v3` not `github.com/gridscale/gsclient-go`, since it uses gomod already.

**Comment by nvthongswansea** [comment]
@bkircher official doc site: https://pkg.go.dev/k8s.io/client-go#section-readme

**Comment by bkircher** [comment]
> I've spent hours to get this godoc server running locally.

It's just typing `godoc` and opening `http://localhost:6060/pkg/`, really. :slightly_smiling_face:

**Comment by nvthongswansea** [comment]
> > I've spent hours to get this godoc server running locally.
> 
> It's just typing `godoc` and opening `http://localhost:6060/pkg/`, really. 🙂

Didn't work on my machine. Issue link https://github.com/golang/go/issues/15049

**Comment by nvthongswansea** [comment]
The command that finally works is `godoc -goroot /usr/share/go`.

**Comment by bkircher** [comment]
Ah understand :smile: 

I added two commits that add a package description (rendered in godoc Overview section) and fix up a bit of line wrapping.

Anyways, this here is not only about the representation of docs (and that it integrates nicely with Golang tooling) but also about the content. Take 

  http://localhost:6060/pkg/github.com/gridscale/gsclient-go/v3/#StorageCreateRequest

for example (if you start your local godoc). StorageType can be nil but docs don't tell me (the user of the library) what storage type is actually chosen if I omit this value. (:joy:  To be fair, API docs are also very silent about this.)

I think we could improve here a bit over time by giving the programmer more insightful, useful documentation on what actually happens in API calls like the above.

But let's prepare a release. Can't wait to get that sped improvement into gscloud!

**Comment by bkircher** [comment]
Another good example of what could be improved are docs like this:
https://github.com/gridscale/gsclient-go/blob/master/template.go#L41

```go
// TemplateProperties JSOn struct of properties of a template
type TemplateProperties struct {
```

Programmers know that this is marshaled to JSON by looking at the struct. Also this is not really important.  A better comment would include
- what this struct represents
- what it is used for
- or how to use it

**Comment by bkircher** [comment]
I fixed above in 538abc3d15a6c93d5ca21282a35b98161107c06a. Take a look. Also fixed the rest of issues in templates.go

You want to go through the other objects and fix the worst? @nvthongswansea
